com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1403,13 +1407,45 @@ public void NetworkUpdate(NetworkUpdateStage updateStage)
             }
         }
 
+        private void ProcessPendingApprovals()
+        {
+            List<ulong> senders = default;
+
+            foreach (var responsePair in ClientsToApprove)
+            {
+                var response = responsePair.Value;
+                var senderId = responsePair.Key;
+
+                if (!response.Pending)
+                {
+                    HandleConnectionApproval(senderId, response);
```

Comment:
try/catch here?

Reply:
I'm hesitant. What would be the desired effect of adding this throw/catch block?

If `HandleConnectionApproval` were to throw and we'd silently catch it, the error would possibly get re-thrown over and over every frame. Letting it escape would alert the developer to the problem. If we were to not be silent about it, then I don't see the point of catching it.

Also, the code was previously performing `HandleConnectionApproval` without a throw/catch block.

Reply:
The current callsite is in the message handler, which is in a try/catch at `Unity.Netcode.Runtime\Runtime\Messaging\MessagingSystem.cs:269`

Catching and logging any exceptions allows you to continue trying to process other pending joins. I'm not sure how it makes it any harder for users to deal with it than the current implementation - users don't have access to this earlyupdate method and so cant do their own exception handling as is... best they get in any case is a log message about an unhandled exception. With the current implementation we also never process any other joins, and we also break the rest of our EarlyUpdate - namely pumping the transport for messages and processing them. Seems like an exception here is fatal without a try/catch